### PR TITLE
MCR-1874 deprecate MCRConfiguration

### PR DESCRIPTION
**Pull request by yagee-de**
[Link to jira](https://mycore.atlassian.net/browse/MCR-1874).

### CODE REVIEW

**Comment by adlerre** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
- Added 3
           

Complexity decreasing per file
==============================
+ mycore-base/src/main/java/org/mycore/datamodel/metadata/share/MCRMetadataShareAgentFactory.java  -1
+ mycore-base/src/main/java/org/mycore/frontend/servlets/MCRServlet.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/MyCoRe/mycore/pullRequest?prid=5299284&bid=17345245)

**Review by fluetze — APPROVED** [review]

File: mycore-ifs/src/main/java/org/mycore/datamodel/ifs/MCRContentStoreFactory.java [hunk 1]
```diff
@@ -168,7 +167,7 @@ private static void initStoreSelector() {
 
         return EXTENDER_CLASSES.computeIfAbsent(storeID, key -> {
             String storeClass = "MCR.IFS.AVExtender." + key + CLASS_SUFFIX;
-            return MCRConfiguration.instance().getClass(storeClass, null);
+            return MCRConfiguration2.<org.mycore.datamodel.ifs.MCRAudioVideoExtender> getClass(storeClass).orElse(null);
```
Comment (adlerre) [inline mycore-ifs/src/main/java/org/mycore/datamodel/ifs/MCRContentStoreFactory.java:170]: Codacy Issue found: [Unnecessary use of fully qualified name 'org.mycore.datamodel.ifs.MCRAudioVideoExtender' due to existing same package import 'org.mycore.datamodel.ifs.*'](https://app.codacy.com/manual/MyCoRe/mycore/pullRequest?prid=5299284)

**Comment by adlerre** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
- Added 3
           

Complexity decreasing per file
==============================
+ mycore-base/src/main/java/org/mycore/datamodel/metadata/share/MCRMetadataShareAgentFactory.java  -1
+ mycore-base/src/main/java/org/mycore/frontend/servlets/MCRServlet.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/MyCoRe/mycore/pullRequest?prid=5299284&bid=17345245)

File: mycore-base/src/main/java/org/mycore/datamodel/common/MCRXMLMetadataManager.java [hunk 2]
```diff
@@ -326,36 +326,36 @@ public MCRMetadataStore getStore(String project, String type, boolean readOnly)
...
         if (slotLayout == null) {
-            config.set(configPrefix + "SlotLayout", defaultLayout);
+            MCRConfiguration2.set(configPrefix + "SlotLayout", defaultLayout);
         }
-        config.set(configPrefix + "BaseDir", typePath.toAbsolutePath().toString());
-        config.set(configPrefix + "ForceXML", true);
-        config.set(configPrefix + "ForceDocType", objectType.equals("derivate") ? "mycorederivate" : "mycoreobject");
+        MCRConfiguration2.set(configPrefix + "BaseDir", typePath.toAbsolutePath().toString());
+        MCRConfiguration2.set(configPrefix + "ForceXML", String.valueOf(true));
+        String value = objectType.equals("derivate") ? "mycorederivate" : "mycoreobject";
```
Comment (adlerre) [inline mycore-base/src/main/java/org/mycore/datamodel/common/MCRXMLMetadataManager.java:357]: Codacy Issue found: [Position literals first in String comparisons](https://app.codacy.com/manual/MyCoRe/mycore/pullRequest?prid=5299284)

File: mycore-mods/src/main/java/org/mycore/mods/merger/MCRMergerFactory.java [hunk 3]
```diff
@@ -1,54 +1,55 @@
...
+ * Builds appropriate MCRMerger instances for a given MODS XML element.
+ * Since MODS elements need to be compared and merged differently, the factory returns
+ * different merger implementations for different element types.
+ *
+ * MCR.MODS.Merger.default=[Default class to merge MODS, typically MCRMerger]
+ * MCR.MODS.Merger.[elementName]=[Specific implementation by element name]
+ *
+ * @author Frank L\u00FCtzenkirchen
+ */
+public class MCRMergerFactory {
```
Comment (adlerre) [inline mycore-mods/src/main/java/org/mycore/mods/merger/MCRMergerFactory.java:34]: Codacy Issue found: [The utility class name 'MCRMergerFactory' doesn't match '[A-Z][a-zA-Z0-9]+(Utils?|Helper)'](https://app.codacy.com/manual/MyCoRe/mycore/pullRequest?prid=5299284)

**Comment by MyCoRe-Travis** [comment]
Something gone wrong on 1ca0fcbdc41de6fc899e2feac8385935a18be4a6! 
 Please Check https://github.com/MyCoRe-Travis/test_artifacts/tree/5411-pull_request-950

**Comment by adlerre** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
- Added 1
           

Complexity decreasing per file
==============================
+ mycore-base/src/main/java/org/mycore/frontend/servlets/MCRServlet.java  -1
+ mycore-base/src/main/java/org/mycore/datamodel/metadata/share/MCRMetadataShareAgentFactory.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/MyCoRe/mycore/pullRequest?prid=5299284&bid=17345245)

File: mycore-mods/src/main/java/org/mycore/mods/merger/MCRMergerFactory.java (hunk 3 above)
Comment (adlerre) [inline mycore-mods/src/main/java/org/mycore/mods/merger/MCRMergerFactory.java:34]: Codacy Issue found: [The utility class name 'MCRMergerFactory' doesn't match '[A-Z][a-zA-Z0-9]+(Utils?|Helper)'](https://app.codacy.com/manual/MyCoRe/mycore/pullRequest?prid=5299284)

**Comment by MyCoRe-Travis** [comment]
Something gone wrong on 030b4f8597ce26d18da9116cc1792e91abf8a607! 
 Please Check https://github.com/MyCoRe-Travis/test_artifacts/tree/5413-pull_request-950

**Comment by adlerre** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
- Added 1
           

Complexity decreasing per file
==============================
+ mycore-base/src/main/java/org/mycore/frontend/servlets/MCRServlet.java  -1
+ mycore-base/src/main/java/org/mycore/datamodel/metadata/share/MCRMetadataShareAgentFactory.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/MyCoRe/mycore/pullRequest?prid=5299284&bid=17345245)

File: mycore-mods/src/main/java/org/mycore/mods/merger/MCRMergerFactory.java (hunk 3 above)
Comment (adlerre) [inline mycore-mods/src/main/java/org/mycore/mods/merger/MCRMergerFactory.java:34]: Codacy Issue found: [The utility class name 'MCRMergerFactory' doesn't match '[A-Z][a-zA-Z0-9]+(Utils?|Helper)'](https://app.codacy.com/manual/MyCoRe/mycore/pullRequest?prid=5299284)

**Review by kkrebs — APPROVED** [review]

**Pull request merged**